### ISSUE TITLE
Metamask + Trezor pop up window automatically closes without signing

### ISSUE THREAD

**Issue by ChooChooCC** [trezor/connect#784]
When the pop up window shows up for me to sign transactions (the part where there is a green circle waiting to load), the window automatically closes before I can sign anything, and in my metamask activity it says "Failed, send X coin".

**Comment by prusnak** [comment]
Which OS are you using?
Which browser are you using?

**Comment by ChooChooCC** [comment]
Mac and brave. The workaround I found is to downgrade to firmware 2.3.0, and reset the metamask and clear my browser cache, now it works with the older version.

**Comment by prusnak** [comment]
Brave is not supported. Please, try if this works in Chrome.

**Comment by SophieElle** [comment]
I also use the brave browser and had problems confirming a certain transaction for weeks for the same reason.

Then it worked. I used the Trezor suite (beta version) for this.

And now it does not work AGAIN.
What could be the reason for this?

The problem: the metaMask wallet I use is connected to the BraveBrowser. I can NOT open this particular wallet anywhere else because of Trezor, as I do not have the private key! :(

so I urgently need your help for this problem

**Comment by bradwbradw** [comment]
@prusnak  it's also broken in Chrome. I'm also a Brave user, but I just tried connecting the Trezor to Metamask in Chrome so that I can test, but the popup closes immediately before signing, and then the metamask page is stuck on the "Looking for your Trezor" screen

![Screen Shot 2021-05-27 at 3 08 57 PM](https://user-images.githubusercontent.com/4715252/119903482-9d4cee00-befd-11eb-864f-4190c91904ac.png)

Im using:
OS: MacOS 10.15.7
Hardware Wallet: Trezor 1.10.0
Browser: Chrome Version 91.0.4472.77 (Official Build) (x86_64)
MetaMask Version 9.5.5

issue at Metamask:
https://github.com/MetaMask/metamask-extension/issues/10896

**Comment by szymonlesisz** [comment]
@bradwbradw do you have any popup blocker installed like Adblock, privacy Badger etc?

**Comment by bradwbradw** [comment]
nope

**Comment by bradwbradw** [comment]
hey @szymonlesisz , I just disabled adblock, and turned off "pops up and redirects allowed" in chrome, restarted Chrome and tried again, but the problem still persists.  This time, i did see the error "Device call in progress" in the metamask UI

**Comment by prusnak** [comment]
@bradwbradw Please, try the same thing in the Incognito mode of Chrome.

**Comment by szymonlesisz** [comment]
could you pls describe what you are doing step by step, bc. im not able to reproduce the issue.
From what i understand you are able to "connect hardware wallet" but you are not able to "sign transaction", is that right?
are you trying to send regular ETH or ERC20 token?

**Comment by bradwbradw** [comment]
I am unable to connect the hardware wallet again after I had removed it
while trying to work around the issue.

@prusnak When I click on "connect hardware wallet" in incognito mode, it just opens up a new window with the
`chrome-extension://.../home.html#new-account/connect` url in regular Chrome, and the same problem happens.

@szymonlesisz for steps, go to: https://github.com/MetaMask/metamask-extension/issues/10896#issue-860433243

**Comment by szymonlesisz** [comment]
i guess this will require additional debugging so i need to ask you to do the following:
1. go to `chrome://extensions/`
1. disable MetaMask extension
1. enable "developer mode" (top right corner)
1. enable MetaMask, now you should see additional option "inspect views", click on "background.html"
![Screenshot from 2021-05-28 15-36-17](https://user-images.githubusercontent.com/3435913/119992186-8fb16a00-bfca-11eb-82ad-1b33f06a9f9a.png)
1. in opened console write: `window.__TREZOR_CONNECT_SRC='https://connect.trezor.io/8/';` This will enable additional logs from trezor-connect.
![Screenshot from 2021-05-28 15-39-28](https://user-images.githubusercontent.com/3435913/119992520-edde4d00-bfca-11eb-8db4-616582a9b555.png)
1. try to "connect hardaware wallet", you should see a lot of logs like:
![Screenshot from 2021-05-28 15-42-42](https://user-images.githubusercontent.com/3435913/119993048-70ffa300-bfcb-11eb-9ac6-1c8f641d449a.png)
1. right-clik on the logs and "save as", then DM me the result (szymon.lesisz@satoshilabs.com) dont worry it should not contain any private data at this point

**Comment by bradwbradw** [comment]
alright, I DM'd you the logs.

>

**Comment by faxotherapy** [comment]
Does anyone know a workaround to this issue? Because, in the mean time, Trezor no longer works with MetaMask. Pop-up window always unexpectedly closes before allowing me to do anything on Trezor. That's the huge impediment as it prevents me from interacting with any DeFi app. Again, on Polygon, issue still persits. I'm not keen at all at moving my funds on MM. Thank you.

**Comment by ChooChooCC** [comment]
@faxotherapy the workaround I found is that I downgraded the firmware 2.3.0 (for model T) and it works. I have been using this firmware version for a few month and it works fine.

**Comment by bradwbradw** [comment]
Using a Trezor One firmware v1.10.0, I'm able to use it with Metamask on
Windows 10 Firefox, but not at all on Mac Catalina (Chrome or Firefox)

I also tend to change networks frequently too, between Matic, Binance Smart
Chain, Fantom and Eth.  I don't know if this has anything to do with the
problem.

**Comment by onetwoitsthewu** [comment]
I am also having major issues with my Model T (firmware 2.3.6) with Metamask. If/when the popup does appear after making transaction request, it tells me that my Trezor device is inactive and that it is being used by another application. When I click on activate, either the pop-up will close and give me the "device was disconnected" error or it will forever spin the "loading" message and then time out saying "unable to claim interface." other times, the transaction just gets stuck in MM, but not appear as pending on etherscan so i have to manually "reset account" in MM to get rid of the stuck txns. is this a firmware issue, as this just started happening recently? any ideas on how to remedy? all of my crypto is effectively stuck at the moment which is no bueno.

**Comment by darkwing** [comment]
I was seeing my Trezor popup window closing upon an attempt to send today, with the following error:  "Forbidden key path"

A quick Google search shows it appears it could be a firmware issue, but I haven't followed the provided steps to ensure they work.

*Edit* Removed link that I've not verified works to fix the issue.

**Comment by prusnak** [comment]
> A quick Google search shows it appears it could be a firmware issue, but I haven't followed the provided steps to ensure they work.

It's not a firmware issue. It's an issue of a caller (Metamask) using the wrong path for selected coin (i.e. not using m/44'/60' for Ethereum or m/44'/1' for any of the Ethereum testnet networks).

**Comment by szymonlesisz** [comment]
I'm really trying to reproduce this but with no luck.
i'm just able to connect every time. there must must be something specific in your configuration and i don't know what it is...

my setup:

macOS Catalina 10.15.15
Metamask 9.5.9
Chrome 91.0.4472.77 - with and without adblock + privacy badger enabled
Brave 1.25.70 (chromium 91.0.4472.77) 

Trezor 1.10.0 and 2.3.6
Trezor Bridge 2.0.27

i even tried it with older setups: older browsers, older MM extensions, older firmwares, custom firmwares, webusb - before chrome 91...

@trezor/qa pls help me out here

**Comment by ChooChooCC** [comment]
@szymonlesisz Did you try reproduce that in alternative network other than ethereum mainnet like BSC, xDai, or Matic? I remember that's when it was failing for me.

**Comment by prusnak** [comment]
> @szymonlesisz Did you try reproduce that in alternative network other than ethereum mainnet like BSC, xDai, or Matic? I remember that's when it was failing for me.

Then you are just confirming my words from above:

> It's not a firmware issue. It's an issue of a caller (Metamask) using the wrong path for selected coin (i.e. not using m/44'/60' for Ethereum or m/44'/1' for any of the Ethereum testnet networks).

Each coin has assigned path - https://github.com/satoshilabs/slips/blob/master/slip-0044.md - and no other path should be set.

I am almost confident it is this issue, because downgrade to 2.3.0 (where we did not check the paths) seems to fix the issue.

Metamask should catch error from Connect and show it in their UI, so it seems there are two issues with Metamask:
* they are not using the correct paths for coins they use
* they are not showing the error Connect returns ("Forbidden key path")

**Comment by faxotherapy** [comment]
MetaMask has gone too far on this issue. I will never ever again use MetaMask with Trezor. It simply is dangerous. It's totally unreliable. I'm currently in a leveraged position on Instadapp and I **cannot unwind it**, because I'm not offered to confirm on my Trezor after clicking on the Confirm button on MM pop-up window.

**Comment by szymonlesisz** [comment]
Instead of downgrading the firmware you can try to temporary disable safety checks mentioned by @prusnak to see if that helps. Trezor will warn you about using invalid slip44 path. 
This may help us to narrow down the issue which networks are using invalid paths in metamask.

1. go to https://suite.trezor.io/web/settings/device/
2. click on "edit" safety checks
![Screenshot from 2021-06-10 06-04-47](https://user-images.githubusercontent.com/3435913/121463242-fd3d9d00-c9b1-11eb-8e88-b189f4c924e2.png)
3. set to "prompt" and confirm on Trezor
![Screenshot from 2021-06-10 06-05-27](https://user-images.githubusercontent.com/3435913/121463280-12b2c700-c9b2-11eb-9df1-ecbddb863323.png)

**Comment by faxotherapy** [comment]
> MetaMask has gone too far on this issue. I will never ever again use MetaMask with Trezor. It simply is dangerous. It's totally unreliable. I'm currently in a leveraged position on Instadapp and I **cannot unwind it**, because I'm not offered to confirm on my Trezor after clicking on the Confirm button on MM pop-up window.

I'm going to reply to myself. The issue seems to happen erratically. When it happens, one should try any combination of the following: close dApp tab/window (or quit/restart Firefox), Lock MM (then re-open it), switch network back and forth (e.g. Ethereum ⇌ Polygon), unplug/replug Trezor (if necessary, connect Trezor directly to USB port), clear cookies/cache for the dApp tab/window, sometimes opening MM in a tab first, and interacting with the MM tab instead of the MM pop-up window help.

The thing I understand is why any two or three of the combination above help solve the issue, at least temporarily.

**Comment by faxotherapy** [comment]
> Instead of downgrading the firmware you can try to temporary disable safety checks mentioned by @prusnak to see if that helps. Trezor will warn you about using invalid slip44 path.
> This may help us to narrow down the issue which networks are using invalid paths in metamask.
> 
>     1. go to https://suite.trezor.io/web/settings/device/
> 
>     2. click on "edit" safety checks
>        ![Screenshot from 2021-06-10 06-04-47](https://user-images.githubusercontent.com/3435913/121463242-fd3d9d00-c9b1-11eb-8e88-b189f4c924e2.png)
> 
>     3. set to "prompt" and confirm on Trezor
>        ![Screenshot from 2021-06-10 06-05-27](https://user-images.githubusercontent.com/3435913/121463280-12b2c700-c9b2-11eb-9df1-ecbddb863323.png)

Don't really know whether the suggestion is addressed to me or not. If so, I'm a bit scared to break everything in my Trezor with these steps. Is it some kind of a debugging process for Trezor?

**Comment by bosomt** [comment]
I was able to reproduce and solve issue via Suite 21.6.1 

As soon you change device setting via https://github.com/trezor/connect/issues/784#issuecomment-858283219 you can send transactions.
Metamask was reporting `Forbidden key path` browser console.

**Info:**
 - Suite version: desktop 21.6.1 (0ce310eb30cff6396b91939d3293b34e8e50d7c1)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4_0) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/21.6.1 Chrome/89.0.4389.69 Electron/12.0.0 Safari/537.36
 - Device: model T 2.4.0 regular

**Comment by STew790** [comment]
I can confirm what @bosomt wrote. With disabled safety checks, I was able to send a transaction. With enabled safety checks, i got the same error message.

**Info:**
 - Suite version: web 21.6.1 (0ce310eb30cff6396b91939d3293b34e8e50d7c1)
 - Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36
 - OS: Linux x86_64
 - Screen: 1920x1080
 - Device: model T 2.4.0 regular

**Comment by szymonlesisz** [comment]
@faxotherapy it was addressed to anyone having this issue 

don't worry you will not break anything.
instead of receiving error (Strict full security) you will be prompted, asked for confirmation on device that the app (metamask) is trying to use incorrect/unknown derivation path (not assigned to requested coin by [bip44 standard](https://github.com/satoshilabs/slips/blob/master/slip-0044.md))

![Screenshot from 2021-06-10 15-25-16](https://user-images.githubusercontent.com/3435913/121533780-e5d9d080-ca00-11eb-9d8e-2d097ec4e5bd.png)
![Screenshot from 2021-06-10 15-27-05](https://user-images.githubusercontent.com/3435913/121533791-e8d4c100-ca00-11eb-9798-9d5e6debff5a.png)


like @prusnak said:
downgrading to 2.3.0 helps only because this "safety check" was not implemented back then and even incorrect paths was allowed

**Comment by szymonlesisz** [comment]
of course it would be better to have "safety checks" enabled (you can enable them back after you send your coins)

but since your assets are locked in metamask, this is the way how to work with them
again following @prusnak 

```
Metamask should catch error from Connect and show it in their UI, so it seems there are two issues with Metamask:

    they are not using the correct paths for coins they use
    they are not showing the error Connect returns ("Forbidden key path")
```

**Comment by bradwbradw** [comment]
> > It's not a firmware issue. It's an issue of a caller (Metamask) using the wrong path for selected coin (i.e. not using m/44'/60' for Ethereum or m/44'/1' for any of the Ethereum testnet networks).
> 
> Each coin has assigned path - https://github.com/satoshilabs/slips/blob/master/slip-0044.md - and no other path should be set.
...
> * they are not using the correct paths for coins they use
> * they are not showing the error Connect returns ("Forbidden key path")

@prusnak, thanks for this resource. I'm just curious as to why it would work erratically (or in my case, it is working fine in Windows with Firefox) if MM is making calls using the wrong derivation path? How could it ever work if Metmask uses the "incorrect path" for an Ethereum-based coin?

I'm still learning here so bear with me... but I would have assumed the same path is used for any network that supports the Ethereum standard.  If that's not the case, how does MetaMask know what path to use, if the only configurable difference is basically the RPC url and the Chain ID?

The BIP44 doc describes using different addresses for each coin, but I believe most Metamask users use the same address across ETH, BNB, MATIC, etc... that's kind of the whole point of cross-chain defi, and it's how tools like xpollinate and multichain works for transferring funds between coins.

**Comment by bradwbradw** [comment]
I just tried right now again with my Mac / Brave setup and I'll just describe exactly what I'm doing, hopefully it helps some developers working for Trezor or Metamask.  I had already removed Metamask entirely, so this is a "clean slate" for repro steps purposes.

- install Metamask, select "create new wallet"
- skip sending tracking data, skip saving key phrase
- add Trezor hardware wallet (it worked as expected)
- when exporting public keys, do not check "remember" (one time a couple months ago I had checked that and it caused a bunch of problems, so now I just avoid it)
- add MATIC custom network (as per https://docs.matic.network/docs/develop/metamask/config-matic/)
- go to https://polywhale.finance/tanks and harvest KRILL (it worked as expected the first time)
- try again to harvest more KRILL from another tank (it fails because the trezor-connect popup disappears right away)
- Metamask shows a pending transaction, but when I click on the link, it goes to `https://explorer-mainnet.maticvigil.com/tx/undefined` so the transaction actually was never created in the first place, because there was some error and it got swallowed up

I came here to try out the workaround, but I think it's only for Model T, whereas I'm using Trezor One.  I don't see "edit safety checks" anywhere at https://suite.trezor.io/web/settings/device/

I'm now going to try to harvest krill one more time to see what the error is in trezor-connect (following https://github.com/trezor/connect/issues/784#issuecomment-850435249)  When I opened background.html, i see 

![Screen Shot 2021-06-10 at 7 14 19 PM](https://user-images.githubusercontent.com/4715252/121621068-4a585800-ca20-11eb-999b-6f6b05a2f3a3.png)

When I try to harvest more krill, the popup disappears before signing, but there are no more log events in this window.

**Comment by szymonlesisz** [comment]
> (...) it is working fine in Windows with Firefox
> (...) How could it ever work if Metmask uses the "incorrect path"
> (...) whereas I'm using Trezor One. I don't see "edit safety checks" anywhere

@bradwbradw 
We are dealing with different configurations/cases here but i'm pretty sure that the reason is still the same -> MetaMask is not handling errors well (or not handling at all) in their UI.

In your case (Trezor 1.10.0) there are no "safety_checks"[1] -> you are getting error different than "Forbidden key path", it may not even be related to trezor-connect as i saw in your logs, but this error is interrupting the process and it's not handled/shown in MetaMask UI.
I suppose macOS an Windows browsers reacts differently for not handled errors. Windows is ignoring them while macOS interrupts running code (?) just a guess.

"Device used in another window" error in connect means that some app opened USB port to read/write but didn't close it properly (interruption) and connect is thinking that different app is talking to Trezor right now.


_[1] "Safety_checks" were added in 1.10.1, released on wednesday, if you update your FW they will be there_

**Comment by szymonlesisz** [comment]
> how does MetaMask know what path to use, if the only configurable difference is basically the RPC url and the Chain ID

they don't and this is why they are using invalid path. for example mentioned `MATIC` according to slip44 should be used with path `m/44'/966'/0'` and chainId `137` but its used with `ETH` path  `m/44'/60'/0'` and chainId `137`

example:

address derived from `m/44'/966'/0'/0/0` (matic) is `0x245d740739078b7A87131C0d1b7020D1BE2Fd971`
while address derived from  `m/44'/60'/0'/0/0` (eth) is `0xc03F9C5A337225Db085947cDD8e221a30A42079A`

from what i saw MetaMask is using the same addresses for every possible network (predefined or custom) which is quite unfortunate, because other wallets which are respecting slip44 will derive those addresses differently -> will not show you your `MATIC` balance since its sent to a different/unexpected address


like for example mycrypto.com
![Screenshot from 2021-06-11 11-51-33](https://user-images.githubusercontent.com/3435913/121668362-8258ad00-caab-11eb-91d2-ba81109c1a44.png)

**Comment by szymonlesisz** [comment]
> try again to harvest more KRILL

`KRILL` is another story, there is no `KRILL` here:
https://github.com/ethereum-lists/chains
or here:
https://github.com/satoshilabs/slips/blob/master/slip-0044.md

so technically `KRILL` is not supported. Maybe this is the reason why signing `KRILL` transaction fails and even if MetaMask is showing you that it is pending in reality it's not there.

Again, looks like the same issue: MetaMask didn't handle signing error and assuming that was signed properly is showing you tx as "pending"

**Comment by bradwbradw** [comment]
this problem went away for me when i updated OS to Big Sur

**Comment by mrkaqz** [comment]
Hi, I am also have very difficult to approve spend on the Metamask + Terzor model T.

The popup is just show flash and close without asking me to export public key or do anything.
I have to close browser and repeat the process again until it randomly work sometime.

Anyone can help me on this?

**Comment by aloceros** [comment]
I have reproduced this issue across multiple devices.  I also have devices where it never occurs.  There is something configuration specific.  But I can confirm that it is definitely not related to the safety checks.  Those reporting safety checks have a different issue.  The problem of the popup closing immediately does not trigger any error in the console.

**Comment by aloceros** [comment]
I have reproduced this issue on a vanilla Chromebook.  Literally ChromeOS + MetaMask + Trezor is the entire setup.  If the bug occurs in this environment I don't think it can be blamed on the configuration of other software.

**Comment by markg31** [comment]
I have the same issue, my eth are stuck on metamask/trezor wallet. I' m using trezor one and windows 10, anyone have solved the issue yet?

**Comment by HeroHann** [comment]
As the other issue was closed...
Is Trezor working with Metamask to solve this? @prusnak  And or should I open an issue for metamask and link this?

**Comment by prusnak** [comment]
> Is Trezor working with Metamask to solve this? @prusnak And or should I open an issue for metamask and link this?

Reported here: https://github.com/MetaMask/metamask-extension/issues/10896

**Comment by meckin** [comment]
This has been happening to me as well for months, extremely annoying.

**Comment by mickmastermike** [comment]
Hi, I am having the same issue. Metamask linked to my Trezor Model one! 
When i try and sign a transactions 99% of the time the Trezor pop up window appears for some 2 seconds and disappears again. I have tried everything i could, reinstall metamask, updated everything to the latest versions, cleared the caches, deleted browsing history etc. A few months back it did this some 1/10 times,and after a log out of metamask and sometime a pc restart the issue was solved. But it appears to be getting worse and worse, sometimes it takes me almost 2 hours just to approve 2 transactions, and as you can imagine trying to claims rewards on various farms etc, its a rather tiring task to perform. Any help on the matter would be most appreciated. Cheers! M

**Comment by lawrenzpng** [comment]
Hi, I'm having the similar experience with my Trezor model one for few weeks now, unable to sign for any transaction via my Metamask wallet. It was working fine a month ago but now once I get to the screen to key in passphrase and confirmation screen, nothing happen. Previously there's one more screen to click proceed but this time that screen is missing and I was left with no respond whatsoever. I'm not sure if there's any pop up window closes within split second that i'm not aware of, I'd tried turning it off to test but without any luck.  Appreciate if there's further update on this issue as all my assets were stuck. Thanks.

**Comment by meckin** [comment]
Same here, it's become unbearable, I'm having to repeat the steps listed above to fix the issue at least a few times a day right now. 

I've had to move to Ledger, it came unbearable.

**Comment by lostintime101** [comment]
Suffering exactly these issues also. Works sometimes and other times not. Restarting can help but not always. The popup flashes for a second and then is gone. All attempts to recover it fail and you're just left hanging with a failed transaction and 70 dollar gas fees charge. 

Newish Windows 10 laptop with Chrome, Trezor One + Metamask.

This is such a terrible experience that I created an account here just to leave this comment. It's that frustrating.

**Comment by alexruzenhack** [comment]
I have had the same issue with Chrome 95.0.4638.??. After restarting to update the Chrome version to 95.0.4638.54 it worked. Then I sent all the tokens to another wallet on MetaMask without connection with Trezor model T. For now, I will be far from this setup. However, for BSC it works like a charm.

**Comment by faxotherapy** [comment]
Well, in my case, I decided to buy a Ledger. Not at all a fan of this company, but decided to be more humble, put my pride out of my way, and buy their product so that I can use it with MetaMask. Too bad for Trezor company. Their must be a reason on why the MM team doesn't do a lot to mitigate things or is relunctant to fix this issue. So, I'm not totally blaming MM team.

However, I'll make sure to check MetaMask competitor, such as XDEFI wallet, so that I can still use my Trezor devices.

Disappointed that MM team does nothing to address the issue, but I'd understand them if they had issues communicating with Trezor team. Trezor company seems to be one of the very best in terms of tech skills, but when it comes to communication, they're not good at all.

**Comment by danny-molamola** [comment]
Got the exact same issue.  I've got a Trezor One and Trezor T, both got this issue intermittently.  Often times I have to unplug and replug or close all the browsers and start again.  Tried a couple of Macbook and tried Chrome and Brave, both are the same.

**Comment by camus1612** [comment]
It's happened to me also. Can't find a solution.
I have Trezor one with chrome.
It's sad to see that this problem is so common and stoshilabs don't come up with any solution.

**Comment by bosomt** [comment]
**QA OK**

MetaMask developers solved this issue in latest 1.8.0 version
tested 10+ times and it works now flawlessly

**Comment by bosomt** [comment]
@ChooChooCC please confirm and close issue

**Comment by prusnak** [comment]
Everyone, please make sure you are running Metamask 10.8.0. They are slowly pushing the update to you which should fix the issue.

@camus1612 @danny-molamola @faxotherapy @alexruzenhack @lostintime101 @lawrenzpng @mickmastermike @HeroHann @vying1 @darkwing @ChooChooCC @bradwbradw @onetwoitsthewu

**Comment by camus1612** [comment]
I upgraded. Worked good at start but now again on Pancakeswap it's still happening.

**Comment by GeraldWallace** [comment]
I upgraded, worked for about 10 txns before the problem persisted consistently. Tried everything in this thread. No solution in sight. 

Also created this account just to comment. It's stopping from making so many essential transactions.

**Comment by danny-molamola** [comment]
I upgraded to 1.80, and it worked most of the time.  Occasionally the issue still happens, but it is much less frequent.  However, it created another issue, when I try to use it on Avalanche with MM and Trezor, all transactions failed with the error "String Overflow".  It worked before the upgrade.  Now I have to use Trezor on another machine, with an older MM version for my Avalanche transactions.

**Comment by aloceros** [comment]
Just a suggestion for everyone having problems - you can use Rabby instead of Metamask.  Since switching my Trezor usage has been trouble free on all EVM chains.

**Comment by danny-molamola** [comment]
Thanks for the suggestion, tried a few transactions using Rabby on Avalanche and it works great.  The only issue on Rabby for me is the lack of Harmony support.

**Comment by danny-molamola** [comment]
Upgraded to 1.81, and the pop up window automatically close issue came back again, see this issue every 2-3 transactions, I'm using it on Harmony and can't use Rabby as alternative.

**Comment by inthernar** [comment]
I started having the above described issue:

Chrome Version 96.0.4664.110 (Official Build) (x86_64)
MM v10.8.1
Trezor T v2.4.3

I am trying to execute a transaction on Matic. The first one or two times the pop-up opens and I can interact with it to complete the transaction. After two or three tries, the pop-up displays and closes before I can interact with it.
On my Trezor it asks me to enter my passphrase in the window. Problem is, the window is gone.
When I check my Trezor Suite it tells me that the Trezor is incorrectly waiting and resets the passphrase to normal on the Model T screen. And the transactions fail.

In order to get it to work again, I have to close out entire from Chrome. Once it reloads, I can try again. It works the first two or three times then back to closing pop-up loop.

Note: I was having this issue prior to upgrading my firmware to 2.4.3 (it was the version right before this one, 2.4.1 maybe, I cannot remember) I was hoping upgrading would have resolved this.

No pop-up blocks, nothing fancy. I read all the comments above, tried many of the things. DevMode is too much for me right now as I am overloaded and super busy. Read several other googled pages with similar issues. This issue is far from closed and happens a lot. I do not have my PC here, so I cannot test with that. I am stuck on my Mac. Every day that passes, I lose a lot of money with the projects that I am in. Please do not take this too lightly, and I am patient. Thank you so much.

Please help.

**Comment by camus1612** [comment]
Guys reopen this, you can see this problem is still valid.

**Comment by bradwbradw** [comment]
Still happening for me too after the update, intermittently.

It happens more often when I have many browser tabs open and things are
lagging. Not sure whether it's due to high CPU or high RAM usage, but the
slowdown effect seems to increase the likelihood of this issue appearing.

**Comment by Ankarrr** [comment]
I have the problems again and again. Can dev please do something to fix this?

**Comment by faxotherapy** [comment]
I came to the conclusion that Trezor devices are utterly useless for DeFi plays. They're only great for storing BTC, ETH, etc. As a result, I decided to go with Ledger for DeFi. In the end, one must choose between a company that's utterly bad in terms of customer support (SatoshiLabs) and a company that's utterly bad with customer data (Ledger).

**Comment by wendys-cats** [comment]
@Ankarrr @bradwbradw @camus1612 @vying1 @danny-molamola thank you for reporting! 

Can you please elaborate on your setup? We are trying to reproduce the issue(s) and this might help narrow it down.

OS; browser type and version; trezor device type and firmware; metamask version; what were you trying to do - network, transaction type, if it was metamask integration with another provider and so on.

**Comment by brbubba** [comment]
> @Ankarrr @bradwbradw @camus1612 @vying1 @danny-molamola thank you for reporting!
> 
> Can you please elaborate on your setup? We are trying to reproduce the issue(s) and this might help narrow it down.
> 
> OS; browser type and version; trezor device type and firmware; metamask version; what were you trying to do - network, transaction type, if it was metamask integration with another provider and so on.

Metamask 10.8.1 
LinuxMint
Brave Browser (Chrome won't even connect to the Trezor)

Swapping 1 FTM to USDC on Spookyswap. Trezor validation window pops up and closes within 2 seconds. Disabling security outlined above has no effect. If you repeat the transaction no window pops up, you have to close out of Brave and relaunch to get the 2 second window to pop up again.

FYI works just fine on windows in Brave. Would prefer to use on dedicated Linux system for crypto that doesn't see daily use.

**Comment by danny-molamola** [comment]
Metamask 10.8.1
I've a MacBook Pro and old Air, both have the same issue
Tried Brave and Chrome, both the same
I have a Trezor One and T, both the same, I use the One more and see this issue more often

Transactions: The issue appears randomly, I saw it on Aave, Curve on Polygon, and DFK on Harmony, and some other sites that I've forgotten.

**Comment by bradwbradw** [comment]
MacOS Big Sur 11.6
Firefox 95.0.2
Trezor One f/w 1.10.4
Metamask 10.8.1
Various networks: eth, fantom, polygon. Any kind of transaction is
susceptible

**Comment by camus1612** [comment]
> @Ankarrr @bradwbradw @camus1612 @vying1 @danny-molamola thank you for reporting!
> 
> Can you please elaborate on your setup? We are trying to reproduce the issue(s) and this might help narrow it down.
> 
> OS; browser type and version; trezor device type and firmware; metamask version; what were you trying to do - network, transaction type, if it was metamask integration with another provider and so on.

Windows 10
Chrome 97.0.4692.71
Trezor One
Firmware: 1.10.4
Metamask 10.8.1

It happens a lot on Pancake Swap on BSC while trying to swap and adding liquidity.
Also happened on Harmony at DefiKingdoms while trying to add liquidity.
On AAVE when trying to repay loan, when trying to deposit collateral.

**Comment by camus1612** [comment]
> I came to the conclusion that Trezor devices are utterly useless for DeFi plays. They're only great for storing BTC, ETH, etc. As a result, I decided to go with Ledger for DeFi. In the end, one must choose between a company that's utterly bad in terms of customer support (SatoshiLabs) and a company that's utterly bad with customer data (Ledger).

I'll take the bad customer support instead. Because if it'll continue, there will be a vacuum that other company will fill. While me staying secure all along.
I hope they will fix the customer support in first priority of course.

**Comment by darkwing** [comment]
For those experiencing this issue, could you do the following:

* Click the "Extensions" icon and choose "Manage extensions": 
<img width="371" alt="ME" src="https://user-images.githubusercontent.com/46655/148557476-7b8ded80-dcd3-4ac4-a574-77fed2a8368b.png">

* Click the "background.html" link, which will open a developer console:
<img width="414" alt="Back" src="https://user-images.githubusercontent.com/46655/148559303-347e740b-da13-4c16-a150-a62b573fc1a5.png">

Please let us know if you see any errors in the console!  Thank you!

**Comment by danny-molamola** [comment]
For every action (click on the buttons on the Trezor pop-up window, or physically press the button on the Trezor One, an error "Refused to set unsafe header 'origin'" is shown on the console.  

![image](https://user-images.githubusercontent.com/51925396/148642215-44561c91-d79c-42d8-8f52-8a3e831e35e7.png)

When the pop-up was closed unexpectedly, there is no new error messages on the console.

As the transaction was stuck at this state, I unplugged the device and it showed the below errors:

![image](https://user-images.githubusercontent.com/51925396/148642191-0ef859ae-3c4c-43cb-b63b-b3e91d011f58.png)

**Comment by inthernar** [comment]
> For those experiencing this issue, could you do the following:
> 
> * Click the "Extensions" icon and choose "Manage extensions":
> 
> <img alt="ME" width="371" src="https://user-images.githubusercontent.com/46655/148557476-7b8ded80-dcd3-4ac4-a574-77fed2a8368b.png">
> 
> * Click the "background.html" link, which will open a developer console:
> 
> <img alt="Back" width="414" src="https://user-images.githubusercontent.com/46655/148559303-347e740b-da13-4c16-a150-a62b573fc1a5.png">
> 
> Please let us know if you see any errors in the console! Thank you!

My screen doesn't have the URLs available, see my attached image, please.

It has started happening a lot more today, like non-stop. I've done the RESET ACCOUNT option. It used to work when I close the browser window and re-open, now six times in a row, the pop-up open and then closes right away. For me, this happens on the MATIC (polygon) network only (so far). I haven't had to use another chain as much as this one.
![Screen Shot 2022-01-08 at 11 39 19 AM](https://user-images.githubusercontent.com/85218428/148659347-2ac11629-6018-4a16-a8e1-7f5bbd412a4f.png)

**Comment by o3o3o** [comment]
![image](https://user-images.githubusercontent.com/4308542/148715594-e6d88188-50f8-44ef-ab51-491bd4c45438.png)
Same error here. It had happened many times :(.....

**Comment by petravis** [comment]
<img width="634" alt="Screen Shot 2022-01-10 at 7 58 23 PM" src="https://user-images.githubusercontent.com/18227370/148889374-02eb1b90-2fb3-4099-b226-88fdbb440ab9.png">
same for me too

edit:  Works on my model T - model one is having the issue

**Comment by warc0s** [comment]
Same error here, and I bought my trezor one a few days ago.. Im regretting so much

**Comment by wendys-cats** [comment]
Thank you very much for providing more info! **I have FINALLY managed to replicate the pop up issue, so hopefully this comment will be helpful for you all.** 
@mrkaqz @vying1 @markg31 @mickmastermike  @lawrenzpng @lostintime101 @alexruzenhack @faxotherapy  @danny-molamola  @camus1612 @GeraldWallace @danny-molamola @ramseynasr @bradwbradw @Ankarrr @brbubba 

**So, to summarize:**

**1. The pop-up problem** - <s>have not reproduced it recently  and it might be OS+browser+settings dependent. We (QA) have experienced the issue in the past, but (not yet) since the new version came out.</s>  **Forever loading trezor connect for signing transactions AND pop up closing itself  - I have managed to replicate this issue.** @szymonlesisz 
Reproduced on Win10 + Chrome  97.0.4692.71 (Official Build) (64-bit) + webUSB for both adblock present and not present.

**How to avoid this issue:** do not check the opt-in boxes mentioned below, disable adblock for trezor connect site (just to be sure), and install the latest bridge (2.0.31). Then, connect your HW account again. If you encounter the same problem even after taking these measures, please report here immediately, so we can look into it. Thank you again for taking the time to write us the additional data!

The issue started for me once I checked these two:

opt in one
![image](https://user-images.githubusercontent.com/58593111/149248170-f667b17f-48ef-4a1d-bb28-6c04a5d70fbb.png)

opt in two - this one is available only when connecting via webUSB.
![image](https://user-images.githubusercontent.com/58593111/149248185-28e7136e-6172-4151-86ab-b9be3993e974.png)

I was also able to temporarily solve it by disabling adblock for trezor connect site, but it worked only for a short while. It is possible that adblock might play a role for some users.


**2. wrong path derivations** - should be solved with new Metamask version. 

**3. Refused to set unsafe header 'origin'**  - when using T1 or TT on Brave. Apparently, Brave wallet uses funny HD paths, and this happens when you try to connect Trezor, Metamask and Brave when the paths are not compatible. 

@petravis @o3o3o can you confirm this? Or is your setup different? I was only able to reproduce this with Brave, but Metamask itself might also do this for certain tokens. @prusnak ?
![image](https://user-images.githubusercontent.com/58593111/149251816-2640a152-d81e-4252-8e66-2427fe2be544.png)

Just so we're on the same page - these are HD paths offered by Metamask (for ETH testnet). The yellow one is compatible with Brave wallet, the red one is not and will result in being unable to connect properly all three (Metamask, Brave, Trezor) and, of course, being unable to sign transactions. 
![image](https://user-images.githubusercontent.com/58593111/149252401-3b523821-e689-4e05-ab46-5585655bc74a.png)


**4. contract interaction failing** - experienced this myself. Created dum dum token contracts on tROP testnet via remix, injected web3 with metamask. Metamask shows a LOT of 429 errors (too many requests), it might be related, and also explain the randomness.

![image](https://user-images.githubusercontent.com/58593111/148473455-238b15d0-7f07-4051-8aaf-4b43a531e5c9.png)


AFAIK, MEW also offers contract interaction, and I have never experienced any issues with it in the past. It's not as pretty, but it should get the job done.

**5. further party providers** - this might be out of scope, and out of our hands, since it is accessing HW accounts on said party provider via Metamask. 


**Other testing info**
**1. Windows 10 and Windows 11**
Adblock present
Browser Chrome Version 97.0.4692.71 (Official Build) (64-bit)
Browser Firefox 96.0 (64 bit)
Metamask version 10.8.1 (polygon mumbai testnet, binance smart chain testnet, ropsten testnet)
Remix 
Pancakeswap 
Devices T1 1.10.4 and TT 2.4.3
Trezord 2.0.27 and 2.0.31, also tried with webUSB for Chrome

**Outcome**: 
Chrome with Metamask shows a LOT of 429 errors. Did not encounter Refused to set unsafe header 'origin'. Did not encounter pop up issue.
Remix with Metamask - Successfully created, compiled, deployed contract, contract interaction works mostly fine, sometimes the response time took too long. Could not reliably reproduce the issue. Might be tied to 429 errors. 
Integration with pancakeswap - only rudimentary. Was able to access trezor via metamask in pancakeswap production, but they do not offer trezor integration/it does not work on testnet.

I also managed to totally break Metamask extension in Chrome TWICE, had to restart everything to get it to work again. Most likely crashed first time when I tried stuff with pancakeswap - it refuses to stay on testnet and switches back to mainnet. Metamask crashed second time when I was testing trezor connect.
![image](https://user-images.githubusercontent.com/58593111/149168785-2aa368bd-fadc-48a1-b1b4-642d33102d13.png)

![image](https://user-images.githubusercontent.com/58593111/149249319-3adf8c18-1bc1-468e-b7c7-9573511f0026.png)



**2 mac Big Sur**
Adblock present 
Chrome Version 96.0.4664.93 (Official Build) (x86_64)
Brave Version 1.34.80 Chromium: 97.0.4692.71 (Official Build) (x86_64) - tested, but Brave is NOT supported!
Metamask extension 10.8.1 (polygon mumbai testnet, binance smart chain testnet, ropsten testnet)
Remix
Devices T1 1.10.4 and TT 2.4.3
Trezord 2.0.27, also tried with webUSB for Chrome

**Outcome**:
Observation: Brave, like Metamask, uses different testnet coins. 
Encountered Refused to set unsafe header 'origin' when using T1 or TT on Brave. Did not encounter pop up issue on either browser. Chrome with Metamask shows a LOT of 429 errors.
![image](https://user-images.githubusercontent.com/58593111/149168069-b8c4c025-7c17-445f-9a62-d1a74addaf9b.png)

**Comment by rsimonton** [comment]
This issue (pop-up window appearing momentarily before closing when attempting to complete/sign a tx) has been long standing and has been very annoying but I've lived with it. Recently it's happening more often (or at least it feels that way).  This morning I was able to reproduce it 100% simply trying to sign a transaction on a proposal on https://commonwealth.im/.  Given that it's been happening for ages I can say with certainty that it's not tied to any one version of the specs below as they've all been upgraded over time and the issue has persisted.

T1, firmware: 1.10.4
Mac: High Sierra 10.13.6 (17G6030)
Brave: Version 1.34.80 Chromium: 97.0.4692.71 (Official Build) (x86_64)
MetaMask (NOT Brave Wallet, which is disabled): 10.8.1
Trezor Suite: 21.12.2

I have Brave Shields (ad block) down for the Trezor Suite site.

@wendys-cats "How to avoid this issue: do not check the opt-in boxes mentioned below" -- I've never once checked the box and always click "allow once for this session".

@darkwing (holy shit, it's David Walsh!) "Click the "background.html" link, which will open a developer console -- I did this and went through the process of connecting and then attempting to sign the transaction, saw the pop-up window appear/disappear and... nothing in the dev console. Nothing of interest in network tab.  I cleared cookies and local storage in the dev console and tried again, same experience.  Nothing useful.

Because I actually needed to use my wallet I took the semi nuclear option and force-quit Brave.  This *always* works to temporarily resolve the issue.

FWIW I'm a dev and am fairly familiar with all of this stuff, happy to help on my end however I can. Thank you all for looking into this.

**Update**:  This disappearing pop-up issue returned on the very next tx I tried, on a different site. lol/sob

**Comment by danny-molamola** [comment]
The issue is still happening.  I have switched from Brave to Chrome, and the pop up closed unexpectedly randomly and the frequency is high when I have many tabs open.  The usual fix is either force quit all chrome browsers or unplug the trezor and plug it back again a few times.  There is no error in the developer console when the pop up closed.  It can be easily reproduced as it basically happens every day.

**Comment by inthernar** [comment]
> The issue is still happening. I have switched from Brave to Chrome, and the pop up closed unexpectedly randomly and the frequency is high when I have many tabs open. The usual fix is either force quit all chrome browsers or unplug the trezor and plug it back again a few times. There is no error in the developer console when the pop up closed. It can be easily reproduced as it basically happens every day.

He's correct. It has been happening consistently. I went over the proposed "solution" provided by @wendys-cats 
and it still happens. He is also correct, in saying that the only way to resolve this is to quit Chrome outright, then re-open and try again. Sometimes the issue persists even after quitting outright. But most of the time the quitting and restarting of Chrome solves the problem for a TX or two, then it returns.

There is also nothing magical about having to recreate this problem. Just try a few transactions in a row and it will start happening. It is weird that the window call to open the Trezor page immediately force closes. And I can confirm that there are no easily visible errors in the console window.

**Comment by ryny24** [comment]
This issue is driving me nuts tonight. When will this get resolved?!?!?!?

**Comment by Sheesh-bit** [comment]
This issue still happens, mostly in the Matic/Polygon Mainnet. The trezor will pops up for a sec and will suddenly close making the transaction unable to process. Tried every solution that is stated here in this thread but nothing works.

**Comment by richyboycaldo** [comment]
I've been having this problems for months now. I can't believe it hasn't been fixed.

**Comment by sime** [comment]
This thread has diverged, but if you have hard examples of reproducible cases, please the steps in a manner similar to below. It's important to share exactly which site/application you're experiencing the bug with. Unfortunately, so far I expect it to be a race condition, which is difficult to track down.

Device: Trezor Model T
**Prerequisites:**
1. Setup MetaMask with [Polygon network RPC](https://docs.polygon.technology/docs/develop/metamask/config-polygon-on-metamask/#add-the-polygon-network-manually).
2. Add connect hardware wallet (Trezor) to MetaMask
3. Login on this [NFT Polygon marketplace](https://polygon.nftically.com/login). Requires signing a message on the device.

**Chrome (MetaMask 10.11.3)**
* First attempt: popup appeared and closed immediately
* Second attempt: works as expected

**Firefox (MetaMask 10.11.3)**
* First attempt: spinner of death
* Second attempt: works as expected

**Comment by ANDYVDL** [comment]
Just started having the same issue. It tuned a year old a week ago! Happy birthday!

**Comment by sime** [comment]
@ANDYVDL I'm looking for more detail about this. Could you share which site you are using, currency, exactly which actioned you performed?

**Comment by terrymason** [comment]
> @ANDYVDL I'm looking for more detail about this. Could you share which site you are using, currency, exactly which actioned you performed?

I'm able to produce this any time - roughly 20% of my actions on the trezor are successful.  If I reboot my computer, fire up my browser, I'm able to make a transaction.  After that my transactions will all fail.  I then do a dance of unplugging my trezor and closing / reopening my browser.  It can get one, maybe two transactions before it fails.

The symptoms are that:
1. I take an action that requires approval
2. Metamask comes up asking me to approve the transaction (I do)
3. A trezor window will open asking me to approve the transaction, and will close within 2 seconds.
4. The transaction will stay as unconfirmed in metamask until I unplug the trezor, when it will fail with "device disconnected"

Windows 11 21H2
Firefox 98.0.2
trezor model T
trezor firmware 2.4.3

Example action:
Questing on Defi Kingdoms (harmony)

**Comment by terrymason** [comment]
I turned off safety checks in my trezor suite and tried again.
First transaction was successful
Second transaction launched the approval window, which then auto closed.  No request for input on the trezor device.   Transaction failed.

**Comment by ian444** [comment]
I'm mostly on Polygon and Harmony and have had 2 PC's running so I can use one while the other restarts to try to get a tx through. One is Linux, the other Win10, one running Chrome, the other Firefox or Chrome. One thing I found to help was to first send a transfer of funds to another wallet, but cancel it at the Trezor. This gets the Trezor and Metamask connected and warmed up. Then might get 2 or 3 tx's through.

I did find the ultimate solution though, I moved my accounts over to a Ledger for defi. It's a breath of fresh air to have stress-free transactions again. Should have done it ages ago. Will keep my Trezors on standby hoping for a fix of these popup windows that Ledger does not need or use. This is such a shame as the Trezors are a nice bit of gear.

**Comment by faxotherapy** [comment]
Same for me, I did exactly the same thing. I moved part of my funds to Ledger X in order to deal with DeFi. It's indeed a wonderful stress-free experience. After 2 months of use, I never had a single issue between MetaMask and Ledger. Thumb up to Ledger for creating products that work nicely beyond Bitcoin maximalism.

**Comment by sicalpath** [comment]
Fuck Fuck Fuck this fucking issue drives me crazy!!!!!!!!!!!!!!!!

**Comment by faxotherapy** [comment]
I really sympathize. This bug is truly a nightmare. In my case, I 've been forced to switch from Trezor to buying yet another wallet, i.e. Ledger. Trezor can only be used for long-term storage. Since using Ledger, not a single problem arose with MetaMask (at least on Brave). It's a shame Trezor is in fact not compatible with DeFi. Personal opinion here, but my feelings tell me SatoshiLabs people are Bitcoin maximalists, and hence, despise those who are not. How could it be otherwise since this issue has not truly been addressed yet. After more than a year. Bolstering this is the fact SatoshiLabs is utterly crap in terms of tech support. In one last desperate attempt, I have contacted some crypto media to let them know about this issue (the one raised on this thread), but have not received any reply so far…

**Comment by sime** [comment]
We recently released an update to Trezor Connect which attempted to [squash race condition](https://github.com/trezor/connect/blob/396d65dca40ab87960dff05088a3ace9ba534350/CHANGELOG.md#fixed) we thought might be related.

I am aware the bug still exists, it's difficult to track down because it is not reproducible 100% of the time.

Which is why I ask contributors of this thread to continue to provide as much information as possible. 
e.g. What currency you are using, what action you are performing on what platform. Then maybe we can detect a pattern. 

What would be more advantageous is a platform that supports test coins where the bug exists. E.g. Can I reproduce this bug with Testnet Polygon and Harmony coins? That what we can hammer at it without the fear of loosing coins.

**Comment by mroz22** [comment]
If you could guys please just enter this url into your browser

http://127.0.0.1:21325/status/

And tell me what is your bridge version. My guess is your bridge version is 2.0.27, right?

**Comment by artystern** [comment]
I literally have the same routine and issues... I get a : sentry-install.js:1 MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 "latest" listeners added. Use emitter.setMaxListeners() to increase limit. Error in my console

<img width="926" alt="image" src="https://user-images.githubusercontent.com/17859244/165215380-f940c57f-f8d6-46ba-b384-f831444e4f9e.png">




> > @ANDYVDL I'm looking for more detail about this. Could you share which site you are using, currency, exactly which actioned you performed?
> 
> I'm able to produce this any time - roughly 20% of my actions on the trezor are successful. If I reboot my computer, fire up my browser, I'm able to make a transaction. After that my transactions will all fail. I then do a dance of unplugging my trezor and closing / reopening my browser. It can get one, maybe two transactions before it fails.
> 
> The symptoms are that:
> 
> 1. I take an action that requires approval
> 2. Metamask comes up asking me to approve the transaction (I do)
> 3. A trezor window will open asking me to approve the transaction, and will close within 2 seconds.
> 4. The transaction will stay as unconfirmed in metamask until I unplug the trezor, when it will fail with "device disconnected"
> 
> Windows 11 21H2 Firefox 98.0.2 trezor model T trezor firmware 2.4.3
> 
> Example action: Questing on Defi Kingdoms (harmony)

**Comment by sime** [comment]
@artystern What currency you are using, what action you are performing on what platform.

**Comment by artystern** [comment]
Every transaction, whether on avax, ethereum or matic is doing this.
Usually first one works then second one pop up doesn't appear. Seems like a
systematic problem with trezor that most people are experiencing...

For my examples I'm interacting with Aave on Avalanch or Tokemak on
Ethereum. Happns every transaction (pop up doesn't appear and nothing
happens)

**Comment by DegenDoc** [comment]
Like many here, this annoying issue would pop up randomly for me too. Highly frustrating to say the least!

This is what I did to fix this annoying problem. 

1. Follow the steps at https://www.alphr.com/how-to-disable-ad-block/
2. Using Chrome, go to https://connect.trezor.io/ and click on the Lock icon next to the url to View Site Information. 
3. Click on Site Settings, then to Privacy and Security, scroll down to Pop-Ups and Ads, and set both to Allow. By default, they are set to Block.
5. Finally refresh the page. That's it! 

I restarted Chrome just in case.

For now, it seems to be working for me. Maybe worth a try for you...

**Comment by KyleF0X** [comment]
> @artystern What currency you are using, what action you are performing on what platform.

Hi Im also having this issue using Metamask on Firefox with Trezor T.
Here's how to reproduce on the pulsechain bridge test net.

- First you will need to get some RINKEBY test net eth from faucet https://rinkebyfaucet.com/
- Then you will need to add the pulsechain test net https://pulsechain.com/
- then get some test pulse from faucet https://faucet.v2b.testnet.pulsechain.com/
- Then use pulseChain Bridge to swap tPLS for pPLS https://pulseramp.com/#/bridge

the first half the the swap goes fine, but then when you go to claim you get an error " "Error
Error: Forbidden key path"
![screen shot transactions pulseramp](https://user-images.githubusercontent.com/96126274/168217098-337daaee-df2e-4da6-87da-d9827efcf43d.jpg)

here are some errors from inspecting element
![errors](https://user-images.githubusercontent.com/96126274/168217218-f159f901-b779-4c0c-90f1-62ea3f521efc.jpg)

If i turn safety checks to "prompt" i get the following 
![trezor path](https://user-images.githubusercontent.com/96126274/168217671-729ca672-8ffa-4a74-addc-0683788e7ac9.jpg)

I didnt approve incase it messed things up

**Comment by sime** [comment]
@KyleF0X Thanks for the detailed report. You need to disable safety checks here because you are signing a testnet transaction with a mainnet derivation path. Which is what your Trezor shows when disabling safety checks. Thus it is a 'forbidden key path'.

You can fix that by selecting 'Trezor testnets' in the hardware wallet connection UI of MetaMask.

Just note, in my experience you cannot have two accounts from the same device, e.g. a standard BIP44 wallet and a 'Trezor testnets'. The latest one overwrites the other. You can always overwrite it back once you are done testing.

<img width="365" alt="Screenshot 2022-05-16 at 15 19 12" src="https://user-images.githubusercontent.com/216917/168601683-b9e9eb06-fbfb-4f6f-8e5d-8e733243acda.png">

**Comment by KyleF0X** [comment]
@sime 
Thanks for the reply
>You can always overwrite it back once you are done testing.

 How would that be done? Would i need to re-enter my private recovery keys?

**Comment by sime** [comment]
(for people watching this thread, this comment is off-topic)

> > You can always overwrite it back once you are done testing.
> 
> How would that be done? Would i need to re-enter my private recovery keys?

No need to re-enter anything. You are changing the derivation path of the connected Trezor. The UI of MetaMask is confusing, to do this, you need to add a 'new' hardware device, but since you are using the same hardware wallet, it will _actually_ update the derivation path of the previously connected one.

**Comment by sirloinfurr** [comment]
I'm having this issue with a specific dapp - lyra.finance on optimism network. I was able to sign a transaction on synthetic with my trezor and MM thru optimism. But when I jump into Lyra, the dapps loads very slowly, and the trezor connect pop up window crashes in 2 seconds. I grabbed the url in the popup, and entered it into a new tab, but that didn't get me anywhere

**Comment by danny-molamola** [comment]
From what I observed, it occurred much more frequently on less powerful
machines.  When I tried to sign a transaction, from the activity monitor it
shows 100% CPU utilization when the Metamask is popping up.  If I wait for
around 10 sec to let the CPU go back down to a lower % before signing on
Metamask, often times the Trezor pop-up can show up after that with no
issue.

This issue has been around for so long, and so many people have the same
issue.  Although it appears randomly, I don't think it is super difficult
to reproduce as I had that issue at least a few times basically every
day, and the same for the other folks on this thread, and many more who
didn't know the existence of this thread.  It is amazing that the Trezor
team has no intention to help the community and can't reproduce this
obvious issue on their end.  We can only self-help in this case.

**Comment by sirloinfurr** [comment]
I've been thinking this is the problem as well.  I have an older pc (2016), and everytime the trezor/MM struggles to process, the laptop's fans **always** start whirling loudly.  so i do think it is a hardware issue as well.

**Comment by VandM3i** [comment]
I think I've found solution , I've closed all of Chrome extension (only meatamask left) , and the trezor pop up window work fast and fine.

**Comment by runatyr1** [comment]
It looks what fixed it for me was resetting the USB device bus. 

Using Linux Fedora, had to install usblib development and compile usb-reset (didn't work from the snap store)
```
sudo dnf install libusb-*
git clone https://github.com/ralight/usb-reset.git
cd usb-reset
sudo make install 
sudo install -d /usr/local/bin
sudo install usb-reset /usr/local/bin
lsusb (to get the vendorid:deviceid:
usb-reset vendor_id:product_id
Then go back to metamask and forget/reconnect hardware wallet
```

From what I have seen in this page and my experience, this issue can happen in Windows, Linux, Firefox, Brave, Chrome. So it's OS independent and web browser independent. It looks like a low level OS issue with the USB subsystem that maybe caches or locks the Trezor usb device incorrectly and the usb-reset command fixes it. Disconnect and connecting the Trezor wallet did not work in my case. Also didn't work closing web browsers, clearing history, locking/unlocking metamask, forgetting and adding device in metamask.

Also, it looks like submitting multiple transactions in a queue and trying to approve many too fast triggers the issue and maybe usb-reset clears the cache or lock causing it.

@prusnak you closed this a few months ago but doesn't look solved, there are still many reports lately. Can you reopen or someone from https://github.com/trezor can follow up?

**Comment by Podnikatel131** [comment]
> When the pop up window shows up for me to sign transactions (the part where there is a green circle waiting to load), the window automatically closes before I can sign anything, and in my metamask activity it says "Failed, send X coin".

I have found a solution for the “Confirmation popup appears and closes in a few seconds”. I have had the same problem for more than one year and can´t access my coins on DEFI protocols with metamask.

The solution is easy: Download Microsoft EDGE + metamask + connect to your Trezor. It will say that Metamask is working primarily with Google Chrome and Mozilla Firefox, agree and continue with EDGE and it will be working!

In one WORD: use EDGE and it will be OK!

**Comment by Alexandermerlos** [comment]
I have tried everything suggested on this form and I can only do 2 transactions out of the 20 - 40 transactions I need to do. 

Why have we not found a solution to this? Suggestion for the dev use chatGPT and ask it to write you the right code to solve the issue. Trezor will force their whole community to move to ledger. Crazy how this is not priority for this team

**Comment by sime** [comment]
@Alexandermerlos which browsers have you tried?

Another alternative is to use [Enkrypt](https://www.enkrypt.com/), as it use that latest version of Trezor Connect (v9). MetaMask has yet to update to it.

**Comment by sirloinfurr** [comment]
@Alexandermerlos I do think it's a cpu issue.  @danny-molamola mentioned this back in october 2022.  at the time, I was having the same metamas/trezor issues about 80% of the time on an older laptop (2016) with mid-level specs.  every time I'd try to execute a transaction, my laptop fans would start wizzing at full throttle.  I got a new laptop in Novebmer 2022, with a much faster processor, and every transaction has executed without a problem on the new laptop.

**Comment by b0jkata** [comment]
My fix for Brave is - right click on metamask icon -> Manage extension -> Site access -> On all sites . Then I restarted Brave and everything was OK. Hope this also fix your problems. (The first time I tried, I also turned on "Allow access to file URLs", but then I tried without it, and it was working just fine.) Good luck!

**Comment by csrsc** [comment]
My fix is qute different :)
I have to many wallets extensions installed in Brave, so : 
1. I went to brave://extensions/
2. I've disabled (not uninstalled), Binance Wallet extension.

And everything works great now :)

**Comment by faxotherapy** [comment]
May I suggest trying XDEFI wallet with Trezor? I use Brave with XDEFI. XDEFI works really great for me. Much faster and far more reliable than MM.

Also, with XDEFI I rarely need to adjust gas manually. It's literally “confirm on pop-up window” and “sign afterwards”.

**Comment by csrsc** [comment]
I don't think so.. we should solve MM+Trezor+Brave problem first.
Suggesting new wallet isn't good propositional in my opinion ( whereas 99% of people use MM right now )...
;)

**Comment by qerphrail** [comment]
The same happened a few times last year and the year before last (since 2021). For a while I used it normally, but now it has reoccurred in an irreducible way. It is no longer possible to sign transactions with the Metamask. I'm quite worried because I need to withdraw some coins and now it's impossible. I don't know what else to do. I use Chrome (updated) and a Trezor T (both Trezor Suite and firmware updated). Metamask is updated too. Despite not being a programmer, I use Linux Mint 20.2 Cinnamon. I'm going to present some tasks that I did following the instructions on this page. I emphasize that none of them were efficient:
- removed all extensions;
- tried it in Incognito mode (even in Incognito Metamask operations are compulsorily launched in the regular version of Chrome);
- restarted the Linux OS;
- I used "allow all cookies" in Chrome;
- for the pop-up link "connect.trezor.io/8/popup.html" I enabled the Pop-ups and redirects permission;
- the same failure happens using a second layer like Arbitrum;
- Metamask is quite heavy to load (it's very slow), especially on Uniswap;
- a curiosity that may be relevant is that occasionally it is possible to make approvals at https://www.sushi.com/swap whereas it is impossible to transact with https://app.uniswap.org/#/swap. Reinforcing this item, it is definitely impossible to make an approval at https://app.uniswap.org/#/swap. The pop-up window stays open for approximately 1 second and then disappears.

As I need to use Chrome as a browser I kindly ask for your help to work with Chrome and Metamask.
I noticed that there is an updated version of Trezor Connect which is Trezor Connect (v9). Could I deduce from the connect.trezor.io/8/popup.html popup window that the version I'm using is v8? And if so, is it possible to upgrade to v9?
I will greatly appreciate your help.

**Comment by sime** [comment]
> As I need to use Chrome as a browser I kindly ask for your help to work with Chrome and Metamask.
I noticed that there is an updated version of Trezor Connect which is Trezor Connect (v9). Could I deduce from the connect.trezor.io/8/popup.html popup window that the version I'm using is v8? And if so, is it possible to upgrade to v9?
I will greatly appreciate your help.

MetaMask recently merged Trezor Connect v9, should be released in version v10.25.0. But we don't have any control over that.

**Comment by ss18ping** [comment]
Are there any updates on the issue? I've been experiencing this for years now with the Trezor One, read this and similar threads two years ago, but nothing seems to have changed. This happens regularly on weak machines with 8GB of RAM (Macbooks for example), if the network is congested or there is some other load on the computer, the Trezor popup opens and immediately closes. Some solution is needed so that the Trezor window simply does not close before all the data is loaded. This happens regardless of which browser is used and which extensions are installed, it's a matter of memory usage and load on the computer. I waited for several years that the problem would somehow be solved, but nothing has changed. Too bad everything is still the same.

UPD: I don't have adblock. This happens even in new clean browser.

**Comment by Phileicht** [comment]
Had the same or at least a similar problem. 
Wanted to connect to OpenSea with MetaMask (v10.25.0) to change my settings.
Trezor (firmware 1.11.2, Suite version 23.2.1) pop up opens and before I could sign anything it closed again.
In my case I needed to disable my Wallet Guard extension and restart the browser (Chrome on Win 10).

Maybe this will help someone, but probably not everyone

**Comment by Phileicht** [comment]
Unfortunately the issue started to come back. Disabling extensions does not work this time.

**Comment by ss18ping** [comment]
Yes, disabling extensions does not help. I've been seeing this bug for several years now. Hoped the team would do something. It is necessary that the Trezor pop-up always opens in the next tab, and not in a separate window. And it is necessary that it never closes, and if something goes wrong, it shows an error code, or some other useful information. When the tabs close on their own and after that you have to turn off the wallet and restart the browser - this is a super frustrating thing.

**Comment by sime** [comment]
Previously I mentioned [Enkrypt](https://github.com/trezor/connect/issues/784#issuecomment-1372028640). It uses the latest version of Trezor Connect v9.

We are only tracking issues with Trezor Connect v9. The version that MetaMask uses (v8) is deprecated.

**Comment by Phileicht** [comment]
@sime good to know.
Tried Enkrypt but couldn't find a way to connect it on opensea, even though it says so on their website and within the browser extension.

**Comment by terrymason** [comment]
I believe that @sirloinfurr is correct, and this is a CPU related issue.  I had this issue on an older laptop - tried different browsers, even reinstalled the OS.  It wasn't until I tried a more powerful desktop that I had no more problems.

**Comment by ss18ping** [comment]
Then it turns out that the matter is in very poor code optimization or some kind of leaks. Because in terms of resource consumption - there is no problem to sign a transaction even on an old mobile phone. When the Trezor popup opens, something happens, because of which the computer starts to warm up and the page crashes.

I wrote to the Metamask team, waiting for an answer.

**Comment by sime** [comment]
> @sime good to know. Tried Enkrypt but couldn't find a way to connect it on opensea, even though it says so on their website and within the browser extension.

I find MetaMask and Enkrypt can conflict with each other. I suggest disabling the MetaMask extension temporarily and see if Enkrypt then starts to wake up.

**Comment by sime** [comment]
>  It wasn't until I tried a more powerful desktop that I had no more problems.

And 

> Then it turns out that the matter is in very poor code optimization or some kind of leaks.

This is the definition of a [race condition](https://en.wikipedia.org/wiki/Race_condition). Which is innately hard to track down.

**Comment by Phileicht** [comment]
I removed MM extension from brave browser before I tried using Enkrypt. Unfortunately there is no Enkrypt listed in the 'Connect your wallet' list on opensea and it does not connect automatically. Even tried to use the connect with MM but it just brought me to the MetaMask homepage.

So, since Enkrypt, does not work with opensea and MetaMask-OS-Trezor seem to be a problem, to me it seems that I (we) don't have many options left. Buying a new laptop just to use MM with Trezor is not an option from a financial perspective.

I'll write Enkrypt and ask them if they have an idea and maybe opensea as well.
But as I understood so far, it is not an issue with Trezor but with MetaMask, so I'll write them as well...
Maybe someone is able to fix that.
If not, it would be sad but I guess I'd need to switch to a ledger or something

**Comment by sime** [comment]
@Phileicht Brave is weird, Just tried to use it for the first time. I guess it conflicting with its built-in wallet.

> Buying a new laptop just to use MM with Trezor is not an option from a financial perspective.

Enkrypt works as expected (for me at least) on Firefox and Chrome. No new laptop required.

Edit: Brave started working after re-starting it.

**Comment by Phileicht** [comment]
@sime it seems like in my case it has been an issue with Pocket Universe and the pop up to connect Enkrypt to OS.
Will check if it's working on chrome with the Pocket Universe extension.
And then I'll check if it's working good with Trezor connect. For now I only used the extension wallet.
Would definitely be an option, since I don't connect my Trezor with anything else and with OS only to move NFTs into a hot wallet.
Thanks for the recommendation. 
I'll keep you updated.

**Comment by Phileicht** [comment]
I had a little chat with Enkrypt, they will have a look at the conflict with Pocket Universe. 
But after I connected an empty trezor wallet, it stopped being able to connect to OpenSea. 
Even with disabled Pocket Universe and no other extension in brave as well as in chrome. 
So sometimes Enkrypt is able to connect to OpenSea, sometimes not...so it is like trading one thing that does not really work for another.
To bad though, it does look good.
But just the fact that I'm not able to use any security extensions like Wallet Guard and Pocket Universe with that wallet at the moment, is kind of a no go for me.
So I'll pass for now on Enkrypt, but I'll keep an eye on it and guess I just have to hope that MM will fix that issue before I loose my patience and buy a ledger.

**Comment by jvnovo** [comment]
> 
Metamask version v10.25.0 is running. Trezor Connect v9 not implemented?

**Comment by qerphrail** [comment]
> > As I need to use Chrome as a browser I kindly ask for your help to work with Chrome and Metamask.
> > I noticed that there is an updated version of Trezor Connect which is Trezor Connect (v9). Could I deduce from the connect.trezor.io/8/popup.html popup window that the version I'm using is v8? And if so, is it possible to upgrade to v9?
> > I will greatly appreciate your help.
> 
> MetaMask recently merged Trezor Connect v9, should be released in version v10.25.0. But we don't have any control over that.

@sime Metamask version v10.25.0 is running. Is Trezor Connect v9 implemented?

**Comment by sime** [comment]
Unfortunately, my prediction was wrong. Recently the internal dependency that uses Trezor Connect v9 was released, now it needs to be upgraded into the main repository of the extension: https://github.com/MetaMask/eth-trezor-keyring/releases/tag/v1.0.0

**Comment by qerphrail** [comment]
> Unfortunately, my prediction was wrong. Recently the internal dependency that uses Trezor Connect v9 was released, now it needs to be upgraded into the main repository of the extension: https://github.com/MetaMask/eth-trezor-keyring/releases/tag/v1.0.0

Ok. We are looking forward to this implementation to resolve this issue.
Who can upgrade Trezor Connect v9 in the main extension repository for us?

**Comment by Hannsek** [comment]
@jvnovo Only MetaMask team.

**Comment by Phileicht** [comment]
New Update of MetaMask extension (10.26.1) still uses eth-trezor-keyring: ^0.10.0 and not v1.0.0 which would use Trezor Connect v9.
I guess we will still have to wait for that fix to happen in MetaMask.

For now, my solution, thanks to @sime is to use Enkrypt in Firefox, which seems to work very well so far.
But I only use it to transfer my NFTs on OpenSea to another wallet if I need to sell or to connect with anything, to claim utilities for example. 
Until Enkrypt works alongside with security extensions like Wallet Guard or Pocket Universe, I will only use ist for that purpose.

I've already asked the Pocket Universe devs if they could fix the problem with Enkrypt, but they can't because Enkrypt seems to take control of the entire window.ethereum object, which causes errors on the Pocket Universe side.
I have asked Enkrypt about this, but there is no sign of this being fixed yet.

**Comment by karliatto** [comment]
@Phileicht Could you please describe step by step how do get to that issue? 
There are too many comments in this issue and some of them might be not related with the others. So some clarification will help in reproducing it.

I am trying to reproduce the issue with MetaMask and I am not able to. 

It's possible that once MetaMask updates to `@metamask/eth-trezor-keyring: v1.0.0`   those issues will be fixed but anyway reproducing it before and after might help to make sure we do not see that issue anymore.

**Comment by Phileicht** [comment]
@karliatto 
Sure.
I did test again yesterday after I did an update for Trezor firmware to v1.12.1 and suite to v23.3.2.
After the update it seemed to work, until I did a reboot of my system this morning.
But this happened before.
- Sometimes it worked again (one time or for a few hours/days) and then stopped working again without changing anything.
- Sometimes it worked one time but after log out and log in again the problem came back.
- Sometimes it worked after disabling or removing some extensions (Wallet Guard, Pocket Universe, etc) and restarting the browser, but after reboot, the problem came back again.
- Sometimes it happened after connecting OpenSea to another account in MetaMask and then log out from OS, switch to the hardware wallet account and log in to OS again.

Here is what I did this morning:
1. Connect Trezor to Computer
2. Open Trezor Suite on Win10 and connect Trezor
3. Open Chrome on Win10 with MetaMask Extension
4. Open https://opensea.io/account and connect to MetaMask
5. Log in to MetaMask -> "hot"-wallet account connected
6. Log out of OpenSea
7. Switch in MetaMask to the connected hardware wallet (Trezor) account.
8. Log in to OpenSea/connect MetaMask -> connected to hardware wallet account.
9. Open Settings in OpenSea
10. MetaMask ask to sign -> sign
11. Trezor Connect pop up opens, tries to use Trezor Connect v8 and closes before I can sign anything

Tested with versions:
MetaMask v10.25.0 & v10.26.1
Trezor firmware v1.11.2 & v1.12.1
Trezor suite v23.2.1 & v23.3.2

Also tested with disabled and removed other browser extensions.

**Comment by bosomt** [comment]
**QA OK**


tested on macARM + Chrome + MetaMask **and** Windows 10 + Chrome + MetaMask /latest versions/

_Device firmware_: latest **2.5.3** firmware with Safety checks disabled.

I was able to login via signing on device with two separate accounts with some ETH present on account.
When i tried to sign with empty account popup just disappeared.

**Comment by aspiers** [comment]
I did not used to have this problem in general but since sometime in the last week or so, I'm now experiencing it 100% of the time on 3 different machines.  In contrast MetaMask on Firefox works flawlessly with the `trezord` bridge running.

**Comment by sandbags-1** [comment]
I also have never had this problem for almost two years only started happening several weeks ago in chrome and Firefox, switched to edge which fixed the issue for a week but now is doing the same. Needs to be fixed or will be a mass exodus to ledger

**Comment by nisnevich** [comment]
I'm facing this nearly all the time with Trezor Model One on Chrome and Firefox. Sometimes resetting metamask, cleaning browser cache and reconnecting trezor helps, I'd say 30% of the time, and 70% popup just disappears and I can't sign anything. Very frustrating. I'm on Manjaro Linux, but I see people in this thread have faced the same on Win and macos. 

I'd happily provide any troubleshooting info you may need.

**Comment by Hannsek** [comment]
We are currently working on fix. No need any more info for now.

**Comment by qerphrail** [comment]
Hi team. Can you give us feedback on how the fix for this issue is going?
Impressively, I am unable to transact. Neither restarting the browser, deleting cookies, cleaning browser cache, etc. This glitch happens in any browser. With enough difficulty, previously, one or another transaction took place. As of today, no more.
The tension and anxiety when we need to transact with Trezor is becoming routine. Instead of bringing us security, Trezor is bringing us discouragement.
As it seems to us, Trezor says that the problem is with Metamask. But, the point is that the responsibility lies with Trezor and not with Metamask. The image being damaged is Trezor. It makes sense? For example, if there is a fault with the ZF transmission on a Ford truck, should I go to Ford or ZF for help? Obviously it's Ford, since I bought the product from them.
Please, gentlemen of the Trezor team, do what you can to help us. Call a meeting with the Metamask team. If necessary, pay to make it happen. Make an investment for this to happen and make a repair on behalf of your customers. This problem has been going on for over 2 years and it still hasn't been fixed! Is it conceivable that long?
Please help us as it is impractical to use Trezor.

**Comment by lostintime101** [comment]
Jvnovo's comment is accurate. I also can't get my Trezor to approve any MM smart contract interactions. The pop-ups disappearing was always frustrating issue for years, but now it's completely broken. Computer restarts or changing browsers no longer help. The account is bricked.

And to answer your question Jvnovo, yes, this problem has been going on for years and hasn't been fixed.

**Comment by sime** [comment]
After a huge amount of time MetaMask has finally merged the necessary changes! 

https://github.com/MetaMask/metamask-extension/pull/18302

We are now eagerly awaiting the next MetaMask release.

**Comment by Hannsek** [comment]
This repo is obsolete. Therefore we will archive it. We track all the issues connected to Connect in [Trezor Suite repo](https://github.com/trezor/trezor-suite).

Regarding this issue, we track all issues regarding Connect popup [here](https://github.com/trezor/trezor-suite/issues/4962).
